tests/validators.rs

```diff
@@ -17,3 +17,22 @@ fn both_validator_and_validator_os() {
         )
         .try_get_matches_from(&["app", "1"]);
 }
+
+#[test]
+fn test_validator_msg_newline() {
+    let res = App::new("test")
+        .arg(Arg::new("test").validator(|val| val.parse::<u32>().map_err(|e| e.to_string())))
+        .try_get_matches_from(&["app", "f"]);
+
+    assert!(res.is_err());
+    let err = res.unwrap_err();
+
+    assert_eq!(
+        err.cause,
+        "Invalid value for \'<test>\': invalid digit found in string"
+    );
+
+    // This message is the only thing that gets printed -- make sure it ends with a newline
+    let msg = format!("{}", err);
+    assert!(msg.ends_with('\n'));
```

Comment:
Can you please test the whole help message?

Reply:
Is it OK to test the exact colorization string, or is there a different way you handle that?

Reply:
The string won't be colorized in the tests. Check how the other tests are testing for the help output. There's an utility.

Reply:
Actually, I think that `ends_with('\n')` is correct here because this is the only thing this test is for. There are other tests checking the whole content.

Reply:
If there are other tests regarding the content, shouldn't those tests be modified instead of adding a new one?

Reply:
Yeah, I think so

Reply:
There are no other tests of validator error messages, so we should keep this test.  There are two things I want to ensure:
1.  The cause string is correct (check above on line 30)
2. The overall console message end with a newline (this check).